core/arch/arm/plat-stm32mp1/scmi_server.c

```diff
@@ -0,0 +1,486 @@
+			stm32_clock_enable(clock->clock_id);
+			clock->enabled = true;
+		}
+	} else {
+		if (clock->enabled) {
+			DMSG("SCMI clock %u disable", scmi_id);
+			stm32_clock_disable(clock->clock_id);
+			clock->enabled = false;
+		}
+	}
+
+	return SCMI_SUCCESS;
+}
+
+/*
+ * Platform SCMI reset domains
+ */
+static struct stm32_scmi_rd *find_rd(unsigned int agent_id,
+				     unsigned int scmi_id)
+{
+	const struct scmi_agent_resources *resource = find_resource(agent_id);
+	size_t n = 0;
+
+	if (resource) {
+		for (n = 0; n < resource->rd_count; n++)
+			if (n == scmi_id)
+				return &resource->rd[n];
+	}
+
+	return NULL;
+}
+
+const char *plat_scmi_rd_get_name(unsigned int agent_id, unsigned int scmi_id)
+{
+	const struct stm32_scmi_rd *rd = find_rd(agent_id, scmi_id);
+
+	if (!rd)
+		return NULL;
+
+	return rd->name;
+}
+
+size_t plat_scmi_rd_count(unsigned int agent_id)
+{
+	const struct scmi_agent_resources *resource = find_resource(agent_id);
+
+	if (!resource)
+		return 0;
+
+	return resource->rd_count;
+}
+
+int32_t plat_scmi_rd_autonomous(unsigned int agent_id, unsigned int scmi_id,
+				uint32_t state)
+{
+	const struct stm32_scmi_rd *rd = find_rd(agent_id, scmi_id);
+
+	if (!rd)
+		return SCMI_NOT_FOUND;
+
+	if (!stm32mp_nsec_can_access_reset(rd->reset_id))
+		return SCMI_DENIED;
+
+	/* Supports only reset with context loss */
+	if (state)
+		return SCMI_NOT_SUPPORTED;
+
+	DMSG("SCMI reset %u cycle", scmi_id);
+
+	if (stm32_reset_assert(rd->reset_id, TIMEOUT_US_1MS))
+		return SCMI_HARDWARE_ERROR;
+
+	if (stm32_reset_deassert(rd->reset_id, TIMEOUT_US_1MS))
+		return SCMI_HARDWARE_ERROR;
+
+	return SCMI_SUCCESS;
+}
+
+int32_t plat_scmi_rd_set_state(unsigned int agent_id, unsigned int scmi_id,
+			       bool assert_not_deassert)
+{
+	const struct stm32_scmi_rd *rd = find_rd(agent_id, scmi_id);
+
+	if (!rd)
+		return SCMI_NOT_FOUND;
+
+	if (!stm32mp_nsec_can_access_reset(rd->reset_id))
+		return SCMI_DENIED;
+
+	if (assert_not_deassert) {
+		DMSG("SCMI reset %u set", scmi_id);
+		stm32_reset_set(rd->reset_id);
+	} else {
+		DMSG("SCMI reset %u release", scmi_id);
+		stm32_reset_release(rd->reset_id);
+	}
+
+	return SCMI_SUCCESS;
+}
+
+/*
+ * Initialize platform SCMI resources
+ */
+static TEE_Result stm32mp1_init_scmi_server(void)
+{
+	size_t i = 0;
+	size_t j = 0;
+
+	for (i = 0; i < ARRAY_SIZE(scmi_channel); i++) {
+		struct scmi_msg_channel *chan = &scmi_channel[i];
+
+		/* Enforce non-secure shm mapped as device memory */
+		chan->shm_addr.va = (vaddr_t)phys_to_virt(chan->shm_addr.pa,
+							  MEM_AREA_IO_NSEC);
+		assert(chan->shm_addr.va);
+
+		scmi_smt_init_agent_channel(chan);
+	}
+
+	for (i = 0; i < ARRAY_SIZE(agent_resources); i++) {
+		const struct scmi_agent_resources *res = &agent_resources[i];
+
+		for (j = 0; j < res->clock_count; j++) {
+			struct stm32_scmi_clk *clk = &res->clock[j];
+
+			if (!clk->name ||
+			    strlen(clk->name) >= SCMI_CLOCK_NAME_SIZE) {
+				EMSG("SCMI clock name invalid");
+				panic();
```

Comment:
`panic("SCMI clock name invalid");`

Reply:
oh true! I forgot you recently made changes to get panic logs on non debug configs.
I will fix all of them. Thanks to point out.